docs/design/HealthEndpoint.md

```diff
@@ -0,0 +1,311 @@
+For each field we get the `field.type` object 
++ Condition 1: If the `field.type.kind` is `SCALAR` then this is a primitive column name.
++ Condition 2: If the `field.type.kind` is `NON_NULL` then check if `field.type.ofType.kind` is `SCALAR`. 
+
+We add all those column names which satisfy either of these two conditions to the top `columnNames` array and finally use this to create the graphQL query.
+
+**GraphQL Query**\
+After getting column names for the entity we create the graphQL query payload 
+`query = $"{{{entityName.ToLowerInvariant()}(first: {First}) {{items {{ {string.Join(" ", columnNames)} }}}}}}"
+`. We execute a **POST** query against the GraphQL base URL.\
+CURL Command
+```
+curl --request POST \
+  --url baseUrl/graphql \
+  --header 'Content-Type: application/json' \
+  --header 'User-Agent: insomnia/10.3.0' \
+  --data '{
+	"query": "{UserTable(first: 4) {items { content id }}}"
+}'
+```
+The time it took to execute the above query is the time elapsed for checking the graphQL health.
+
+
+## Output Sample
+After executing the above queries for DB and entities, we create the below sample for comprehensive Health Check Report.
+```
+{
+  "status": "Unhealthy/Healthy",
+  "version": "1.2.10",
+  "app-name": "dab_oss_1.2.10",
+  "dab-configuration": {
+    "http": true,
+    "https": true,
+    "rest": true,
+    "graphql": true,
+    "telemetry": true,
+    "caching": true,
+    "mode": "development"
+  },
+  "checks": {
+    {
+      "name": "database-name",
+      "status": "Healthy",
+      "tags": ["data-source", "mssql"],
+      "data": {
+          "responseTimeMs": 10,
+          "maxAllowedResponseTimeMs": 10
+      }
+    },
+    {
+      "name": "<entity-name>",
+      "status": "Healthy",
+      "tags": ["endpoint", "rest"],
+      "data": {
+          "responseTimeMs": 10,
+          "maxAllowedResponseTimeMs": 10
+      }
+    },
+    {
+      "name": "<entity-name>",
+      "status": "Healthy",
+      "tags": ["endpoint", "graphql"]
+      "data": {
+          "responseTimeMs": 20,
+          "maxAllowedResponseTimeMs": 50
+      }
+    },
+    {
+      "name": "<entity-name>",
+      "status": "Unhealthy",
+      "tags": ["endpoint", "graphql"]
+      "exception": "{exception-message-here}",
+      "data": {
+          "responseTimeMs": 20,
+          "maxAllowedResponseTimeMs": 10
+      }
+    },
+  }
+}
+```
+
+## Test Scenarios
+
+Health Check result scenarios for different cases of Health check and GraphQL and Rest entities enabled or disabled.
+
+> '_' means enabled/disabled (doesn't matter)
+
+* Roles (Global Health and Entity Health are both enabled)
+  * Runtime Health check parameter, `runtime.health.roles` contains "UserRole". However entity doesn't have read permissions for this role.
+    * Health for this entity is displayed with `status: Unhealthy` and `exception: Health could not be checked for this entity as it does not have permissions to perform read operation`. 
```

Comment:
I believe it would better to allow for a 30x http status to be returned and react to that. This makes it so you do not have to look up entity permissions, you call every entity and simply handle them all the same. This enforces permissions without requiring the health check to look them up - which seems far more ideal.